Array_Vector_Exercises/Exercise1/TDinamicArray.h

```diff
@@ -0,0 +1,242 @@
+	T& operator[] (const int Index)
+	{
+		return DynamicArray[Index];
+	}
+
+	const T& operator[] (const int Index) const
+	{
+		return DynamicArray[Index];
+	}
+
+	//-----------------------------------------------------------------------
+	T& Front()
+	{
+		return DynamicArray[0];
+	}
+
+	const T& Front() const
+	{
+		return DynamicArray[0];
+	}
+
+	//------------------------------------------------------------------------
+	T& Back()
+	{
+		return DynamicArray[SizeArray - 1];
+	}
+
+	const T& Back() const
+	{
+		return DynamicArray[SizeArray - 1];
+	}
+
+	//-----------------------------------------------------------------------
+	T* GetData()
+	{
+		return DynamicArray;
+	}
+
+	const T* GetData() const
+	{
+		return DynamicArray;
+	}
+
+	//------------------------------------------------------------------------
+	bool IsEmpty() const
+	{
+		return SizeArray == 0;
+	};
+
+	//-------------------------------------------------------------------------
+	int GetSize() const
+	{
+		return SizeArray;
+	}
+
+	//-------------------------------------------------------------------------
+	int GetCapacity() const
+	{
+		return CapacityArray;
+	}
+
+	//-------------------------------------------------------------------------
+	void Reserve(const int NewCapacity)
+	{
+		if (NewCapacity > CapacityArray)
+		{
+			T* ContainerArray = new T[NewCapacity];
+
+			for (int i = 0; i < SizeArray; i++)
+			{
+				ContainerArray[i] = DynamicArray[i];
+			}
+			delete[] DynamicArray;
+
+			DynamicArray = ContainerArray;
+			CapacityArray = NewCapacity;
+		}
+	}
+
+	//--------------------------------------------------------------------------
+	void ShrinkToFit()
+	{
+		if (CapacityArray != SizeArray)
+		{
+			T* ContainerArray = new T[SizeArray];
+
+			for (int i = 0; i < SizeArray; i++)
+			{
+				ContainerArray[i] = DynamicArray[i];
+			}
+			delete[] DynamicArray;
+
+			DynamicArray = ContainerArray;
+		}
+	}
+
+	//--------------------------------------------------------------------------
+	void Clear()
+	{
+		SizeArray = 0;
+	}
+
+	//--------------------------------------------------------------------------
+	void PushBack(const T& InType)
+	{
+		if (SizeArray == CapacityArray)
+		{
+			Reserve(CapacityArray * 2);
+			DynamicArray[SizeArray] = InType;
+			SizeArray++;
+		}
+		else
+		{
+			DynamicArray[SizeArray] = InType;
+			SizeArray++;
+		}
+	}
+
+	//-----------------------------------------------------------------------------
+	void Pushpop()
+	{
+		if (SizeArray != 0)
+		{
+			SizeArray--;
+		}
+	}
+
+	//-----------------------------------------------------------------------------
+	void Insert(const int Index, const T& InType)
+	{
+		if (Index < SizeArray && Index >= 0)
+		{
+			if (SizeArray == CapacityArray)
+			{
+				Reserve(CapacityArray * 2);
+			}
+				
+			for (int i = 0; i < SizeArray; i++)
+			{
+				DynamicArray[Index + 1] = DynamicArray[Index];
+			}
+			DynamicArray[Index] = InType;
+			SizeArray++;	
```

Comment:
this isn't working as is intended, you are not moving all the values +1 position, and also you are losing one value

![image](https://user-images.githubusercontent.com/109237145/189552985-94ac4956-26a9-4d91-8c28-1ab88964906f.png)

I made this test :p, the resoult should be 71234567890